src/Network/Connection.php

```diff
@@ -597,8 +597,13 @@ private function connectionHasDataToRead($timeoutSec, $timeoutMicros)
                     return false;
                 }
             }
-            time_nanosleep(0, 2500000); // 2.5ms / 0.0025s
+
+            $slept = time_nanosleep(0, 2500000); // 2.5ms / 0.0025s
+            if (\is_array($slept)) {
+                return false;
```

Comment:
what problem does this change fix?
before this change the method tried to read more data in this case.. why doesn't this work?

Reply:
The idea is the same as in #117, but in this case we receive OS signal inside `time_nanosleep`. Function `time_nanosleep` terminates and after that `isDataOnStream` method is called. The `stream_select` function will honestly wait without returning `false`, because signal has been already processed. 
When `time_nanosleep` returns array, this means that it was interrupted by signal. Since here is the data waiting loop, it should be terminated by signal.